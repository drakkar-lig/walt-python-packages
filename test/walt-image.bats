source $BATS_TEST_DIRNAME/common.sh

ERR_IMAGE_SEARCH="
The following command should output '1', it did not:
walt image search test-suite | grep "$TEST_IMAGE_URL" | wc -l
"

@test "walt image show" {
    walt image show
}

@test "walt image search" {
    num_matches=$(walt image search test-suite | grep "$TEST_IMAGE_URL" | wc -l)
    [ "$num_matches" -eq 1 ] || {
        echo "$ERR_IMAGE_SEARCH" >&2
        return 1    # test failed
    }
}

@test "walt image clone" {
    walt image clone --force "$TEST_IMAGE_URL" | grep Done
    image_exists pc-x86-64-test-suite
}

@test "walt image rename" {
    walt image rename pc-x86-64-test-suite $(bats_test_image) && \
    image_exists $(bats_test_image)
}

@test "walt image duplicate" {
    walt image duplicate $(bats_test_image) $(bats_test_image)-2 && \
    image_exists $(bats_test_image)-2
}

@test "walt image cp" {
    test_walt_cp image $(bats_test_image)
}

@test "walt image shell" {
    test_walt_shell image $(bats_test_image)
}

@test "walt image publish" {
    walt image clone --force "$TEST_IMAGE_URL"
    walt image publish pc-x86-64-test-suite
    walt image remove pc-x86-64-test-suite
}

@test "walt image remove" {
    for image in $(walt image show | grep $(bats_test_image) | awk '{ print $1 }')
    do
        walt image remove $image
        image_exists $image && return 1
    done
    true    # ok if we get here
}
