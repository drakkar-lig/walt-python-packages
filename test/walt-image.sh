source $TESTS_DIR/includes/common.sh

define_test "walt image show" as {
    walt image show
}

define_test "walt image search" as {
    walt image search test-suite | grep "$TEST_IMAGE_URL"
}

define_test "walt image clone" as {
    walt image clone --force "$TEST_IMAGE_URL" | grep Done
    image_exists pc-x86-64-test-suite
}

define_test "walt image rename" as {
    walt image rename pc-x86-64-test-suite $(test_suite_image)
    image_exists $(test_suite_image)
}

define_test "walt image duplicate" as {
    walt image duplicate $(test_suite_image) $(test_suite_image)-2 && \
    image_exists $(test_suite_image)-2
}

define_test "walt image cp" as {
    test_walt_cp image $(test_suite_image)
}

define_test "walt image shell" as {
    test_walt_shell image $(test_suite_image)
}

define_test "walt image publish" as {
    walt image clone --force "$TEST_IMAGE_URL"
    walt image publish pc-x86-64-test-suite
    walt image remove pc-x86-64-test-suite
}

define_test "walt image remove" as {
    for image in $(walt image show | grep $(test_suite_image) | awk '{ print $1 }')
    do
        walt image remove $image
        image_exists $image && return 1
    done
    true    # ok if we get here
}
