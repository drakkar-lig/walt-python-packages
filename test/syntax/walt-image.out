
WalT platform control tool
-> management of WalT-nodes operating system images

Usage:
    walt image SUBCOMMAND [args...]

Help about a given subcommand:
    walt image SUBCOMMAND --help

Help about WalT in general:
    walt help show

Sub-commands:
    clone      clone a remote image into your working set
    cp         transfer files (client machine <-> image)
    duplicate  duplicate an image of your working set
    publish    publish a WalT image on the docker hub
    remove     remove an image from your working set
    rename     rename an image of your working set
    search     search for remote WalT node OS images
    shell      modify an image through an interactive shell
    show       display your working set of walt images
